launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -241,9 +241,11 @@
     }
 
     args.addAll(parsedArgs);
+
     if (appResource != null) {
       args.add(appResource);
```

Comment:
When you do `run-example`,  you just specify the class name(e.g. SparkPi),  and no need to specify the app-jar. So in backend code, `appResource` should be auto-find-and-set as the examples' main app jar (e.g. `./examples/jars/spark-examples_2.12-3.0.0.jar`), and then be added as an arg ( this arg will later be set as the so-called `primaryResource`).